src/cmd/uncommit.rs

```diff
@@ -100,9 +100,13 @@ fn run(matches: &ArgMatches) -> Result<()> {
         let target_object = repo
             .revparse_single(commitish)
             .map_err(|_| anyhow!("Invalid commitish `{commitish}`"))?;
+
         let mut target_commit = target_object
+            .peel(git2::ObjectType::Commit)
+            .map_err(|_| anyhow!("Target `{commitish}` cannot be evaluated as a commit"))?
             .into_commit()
-            .map_err(|_| anyhow!("Target `{commitish}` is not a commit"))?;
+            .unwrap();
```

Comment:
I am assuming here that if the call to `peel` is successful that we don't need to check for errors from `into_commit()`. Does that seem reasonable?

Reply:
Yeah, that was ugly. I've changed it.